src/main/asciidoc/repositories.adoc

```diff
@@ -214,6 +214,82 @@ In this first step you defined a common base interface for all your domain repos
 NOTE: Note, that the intermediate repository interface is annotated with `@NoRepositoryBean`. Make sure you add that annotation to all repository interfaces that Spring Data should not create instances for at runtime.
 
 
+[[repositories.nullability]]
+=== Null handling of repository methods
+
+As of Spring Data 2.0, repository CRUD methods that return an individual aggregate instance use Java 8's `Optional` to indicate the potential absence of a value.
+Besides that, Spring Data supports to return other wrapper types on query methods:
+
+* `com.google.common.base.Optional`
+* `scala.Option`
+* `io.vavr.control.Option`
+* `javaslang.control.Option` (deprecated as Javaslang is deprecated)
+
+Alternatively query methods can choose not to use a wrapper type at all.
+The absence of a query result will then be indicated by returning `null`.
+Repository methods returning collection like values will never return null but an empty value.
+
+[[repositories.nullability.annotations]]
```

Comment:
This section duplicates [the section on Null-safety](https://github.com/spring-projects/spring-data-commons/blame/c9e0803beedb7284cc6a51a4a67aa20dd1cb6898/src/main/asciidoc/repositories.adoc#L869) that is currently in place. It would make sense to move `core.nullability-validation` in here.

Reply:
After revisiting the other version of the docs, I'd actually like to keep this one here and augment it with the following aspects currently present in the other version:

- Nullability annotations should list the fully-qualified name at least once, so that it's clear what to import. I think we can (should?) leave out the remark with JSR-305 needed for meta-annotation usage. That's taking it a bit too far I think.
- The `kotlin-reflect` library needed to get the runtime checks enabled.